presto-parquet/src/main/java/com/facebook/presto/parquet/batchreader/decoders/plain/FixedLenByteArrayShortDecimalPlainValuesDecoder.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.parquet.batchreader.decoders.plain;
+
+import com.facebook.presto.parquet.batchreader.SimpleSliceInputStream;
+import com.facebook.presto.parquet.batchreader.decoders.ValuesDecoder.ShortDecimalValuesDecoder;
+import com.facebook.presto.spi.PrestoException;
+import io.airlift.slice.Slices;
+import org.apache.parquet.column.ColumnDescriptor;
+import org.openjdk.jol.info.ClassLayout;
+
+import static com.facebook.presto.parquet.ParquetTypeUtils.getShortDecimalValue;
+import static com.facebook.presto.spi.StandardErrorCode.NOT_SUPPORTED;
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.lang.String.format;
+
+public class FixedLenByteArrayShortDecimalPlainValuesDecoder
+        implements ShortDecimalValuesDecoder
+{
+    private static final int INSTANCE_SIZE = ClassLayout.parseClass(FixedLenByteArrayShortDecimalPlainValuesDecoder.class).instanceSize();
+
+    private final ColumnDescriptor columnDescriptor;
+    private final int typeLength;
+    private final ShortDecimalFixedWidthByteArrayBatchDecoder decimalValueDecoder;
+    private final SimpleSliceInputStream input;
+
+    public FixedLenByteArrayShortDecimalPlainValuesDecoder(ColumnDescriptor columnDescriptor, byte[] byteBuffer, int bufferOffset, int length)
+    {
+        this.columnDescriptor = columnDescriptor;
```

Comment:
I know the existing decoders code has the same pattern, but can you please add requireNonNull() on columnDescriptor and byteBuffer? Same for all other constructors for the classes you add. Thanks!